### ISSUE TITLE
CLOSE_WRITE inotify event when reading metadata with FileRef 

### ISSUE THREAD

**Issue by dbogdanov** [taglib/taglib#756]
I am [reading metadata](https://github.com/MTG/essentia/blob/master/src/algorithms/io/metadatareader.cpp#L211) from the file using FileRef object. For reading only I would expect to see the CLOSE_NOWRITE event in the end, but instead CLOSE_WRITE is output.

**Comment by TsudaKageyu** [comment]
It's the correct behavior, but we have a workaround.

According to the [manpage](https://linux.die.net/man/1/inotifywatch), ```CLOSE_WRITE``` means that the file was opened in writeable mode and does not necessarily imply the file was written to. ```FileRef``` always tries to open a file in read-write mode first.

If you really need to open files in read-only mode, you can create ```TagLib::FileStream``` in read-only mode and pass it to ```FileRef```. Or maybe we should consider #337 again.

**Comment by dbogdanov** [comment]
Indeed, the read-only flag will be the best solution. I was expecting to see it once I've encountered the problem with CLOSE_WRITE.

**Comment by sbooth** [comment]
I think #337 would be a good change to make.  I think it's good practice to open files with the lowest permissions required.

**Comment by michihenning** [comment]
We ran into this too, and it is seriously annoying not being able to open a file read-only. In particular, the spurious inotify events can cause a lot of needless activity in processes that watch for changes to media files. If I only want to read tags, but not modify them, I should be able to open the file read-only.

**Comment by michihenning** [comment]
If compatibility is a concern, why not open the file read-only initially, and re-open read-write the first time an update method is called?

**Comment by TsudaKageyu** [comment]
I see your needs and I think it would be a good thing to do.
However, supporting read-only `FileRef` is not so easy task. I will summarize the problems later on, but we will need to consider the design carefully.

**Comment by flaviotordini** [comment]
Being able to open files in readonly mode is also important for apps in the iOS or macOS sandbox.

**Comment by ufleisch** [comment]
You can use a `FileStream` constructed with `openReadOnly=true`.
